articles/most-useful-node.js-packages/index.md

```diff
@@ -0,0 +1,349 @@
+
+**Request has the following main contractors:**
+-   **Body** - The data to send with the request.
+-   **Cache** - A sting with the following instanceS: default, force-cache, no-cache, no-store, only-if-cached, reload etc.
+-   **Credentials** - A sting which include omit, same-origin instance properties.
+-   **Headers** - It is the `http` headers to send with the request. It is passed to the headers contractor.
+-   **Method** - HTTP method such as `GET` `POST` `DELETE`, `PUT`.
+-   **Mode** - A string with one of same-origin, no-cors, cors instances.
+
+**Examples of simple Request**
+```js
+const request = require('request');
+request('https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY', { json: true }, (err, res, body) => {
+    if (err) { return console.log(err); }
+    console.log(body.url);
+    console.log(body.explanation);
+});
+```
+With request, you need less code to accomplish the same task that you would have used a lot of coding of you were using the HTTP library.
+
+**Other HTTP REST client libraries include**
+-   [`Axios`](https://www.npmjs.com/package/axios) - Used promise-based
+    HTTP client for Node.js and Browsers.
+-   [`Method-override`](https://github.com/expressjs/method-override) -
+    Lets you use HTTP verbs such as DELETE and PUT in places where the
+    client does not support it.
+### 7. JShint
+It is a static code analysis tool to detect errors and potential problems in JavaScript. Lining tools help Node.js developers to analyse common bugs in your code without your focussed attention. Linter analyses the code themselves. JShint scans your Node.js program and reports commonly made mistakes and potential bugs such as syntax errors, leaking variables, implicit type, conventions, etc.
+[`ESLint`](https://eslint.org/) is a similar package example used for linting (enhancing code quality).
+### 8. Morgan
+It is an HTTP request logger middleware for Node.js application. Morgan gives you Insights on how your app is being used and alerts you on potential errors and issues that could be threats to your application. Morgan is considered the most reliable HTTP logger by Node.js developers.
+You can choose to use [`Winston`](https://www.npmjs.com/package/winston) as an alternative library for logging.
+### 9. Karma
+A Node.js developer, you need to do more testing to your application to make sure it's stable and reliable with good performance. Karma is the tool for Job Karma. The main goal of karma is to bring a productive testing environment to developers.
+
+**Why you should choose karma**
+-   Tests code in real browsers.
+-   Test code is multiple browsers(desktop, mobile, tablets).
+-   It tests on real devices such as mobilephone, tablets or even a headless phantomJS.
```

Comment:
Add space between mobile and phone or even better just use mobiles or smartphones.